### ISSUE TITLE
CMake Tools doesn't support source files outside the directory of the  CMakeLists.txt 

### ISSUE THREAD

**Issue by julee** [microsoft/vscode-cmake-tools#1140]
## Brief Issue Summary
e.g:
A workspace has a CMakeLists.txt, and it add a external directory as submodule: 

```
add_subdirectory(externalSubModule path_to_external_submodule_outside_root_directory)
```
While we set: 
"C_Cpp.default.configurationProvider": "vector-of-bool.cmake-tools"

Source files in workspace directory have correct intellisense, but if we navigated to source files of the external submodule by add_subdirectory,  intellisense won't work.

## Expected:
The same behavior as sources files in workspace directory.

**Comment by bobbrow** [comment]
Do the files show up incorrectly in the project outline as well?  I just tried this and it looks like it's concatenating paths.

![image](https://user-images.githubusercontent.com/12818240/77814951-1c813080-7073-11ea-8ea5-77b9620e13f3.png)

**Comment by julee** [comment]
> Do the files show up incorrectly in the project outline as well? I just tried this and it looks like it's concatenating paths.
> 
> ![image](https://user-images.githubusercontent.com/12818240/77814951-1c813080-7073-11ea-8ea5-77b9620e13f3.png)

Yes, external source files paths are wrong. Though we can navigate to source file with "goto definition" .

**Comment by bobbrow** [comment]
Thanks for confirming.  I have a PR out with a fix.

**Comment by andreeis** [comment]
CMake Tools 1.4.0 Beta is available on GitHub.
https://github.com/microsoft/vscode-cmake-tools/releases/tag/1.4.0-beta

Please try it out and let us know if you encounter any issues.

**Comment by julee** [comment]
CMake Tools 1.4.0 Beta works now. Thanks!

**Comment by andreeis** [comment]
This fix is available in CMake Tools #1.4.0 which has been released.
